art/defences/preprocessor/spatial_smoothing.py

```diff
@@ -92,22 +91,33 @@ def __call__(self, x, y=None):
         :return: Smoothed sample
         :rtype: `np.ndarray`
         """
-        if len(x.shape) == 2:
+
+        x_ndim = x.ndim
+        if x_ndim == 2:
+            raise ValueError("Feature vectors detected. Smoothing can only be applied to data with spatial dimensions.")
+
+        if x_ndim > 5:
             raise ValueError(
-                "Feature vectors detected. Smoothing can only be applied to data with spatial " "dimensions."
+                "Unrecognized input dimension. Spatial smoothing can only be applied to image and video data."
             )
-        if self.channel_index >= len(x.shape):
-            raise ValueError("Channel index does not match input shape.")
 
-        size = [1] + [self.window_size] * (len(x.shape) - 1)
-        size[self.channel_index] = 1
-        size = tuple(size)
+        filter_size = [self.window_size] * x_ndim
+        # set filter_size at batch and channels indices to 1
+        filter_size[0] = 1
+        filter_size[self.channel_index] = 1
+        # set filter_size at temporal index to 1
+        if x_ndim == 5:
+            temporal_index = 2 if self.channel_index == 1 else 1
+            filter_size[temporal_index] = 1
+        # Note median_filter:
+        # * center pixel located lower right
+        # * if window size even, use larger value (e.g. median(4,5)=5)
+        result = median_filter(x, size=tuple(filter_size), mode="reflect")
 
-        result = ndimage.filters.median_filter(x, size=size, mode="reflect")
         if self.clip_values is not None:
             np.clip(result, self.clip_values[0], self.clip_values[1], out=result)
 
-        return result.astype(ART_NUMPY_DTYPE), y
```

Comment:
Did you check if after removing this case to `ART_NUMPY_DTYPE` this call still returns an array of format `ART_NUMPY_DTYPE` for all possible input parameter formats?

Reply:
`scipy.ndimage.median_filter` will by default return an array of the same dtype as input.